### DISCUSSION TITLE
Android 12 January update - vpn hotspot uses actual device tethering and not vpn

### DISCUSSION THREAD

**Discussion by sainz95** [Mygod/VPNHotspot#322] [Q&A]
As title states the issue I am having is that on a fresh install of Android 12 stockrom latest Jan 2022 update(rooted via magisk) the vpn hotspot app will detect vpn as tun0 but when "Wi-Fi hotspot" is toggled it uses the default system tethering which is detected by carrier and bypasses the vpn upstream. The IPs from an ip checker website shows carrier IPs and not vpn IP address.

Temporary hotspot toggle does route through VPN but settings change every run which is cumbersome to deal with entering multiple SSIDs/passwords etc when it is turned on/off. Log attached. https://drive.google.com/file/d/10_In_dhCBLmeAB9LRN6r-MIZ6Cqqh4po/view?usp=sharing

Thank you.

**Comment by Mygod** [comment]
Did you disable hardware tethering acceleration?

**Comment by sainz95** [comment]
Yes and closed the app each time after changing settings. Reboot also had
no effect after a change.

**Comment by Mygod** [comment]
Your log seems to suggest that it is not disabled.

**Comment by sainz95** [comment]
Meant to say just now I tried.

**Comment by sainz95** [comment]
It seems that the repeater function works with a desired custom SSID and works over vpn. I must point out that Wifi hotspot shows the android native hotspot icon in the taskbar but repeater shows a custom icon and vpn works with that.

Can you describe the differences between repeater, temporary hotspot, and wifi hotspot?

**Comment by Mygod** [comment]
Their functionalities are as described. Are you sure it worked before January update?

**Comment by sainz95** [comment]
I cannot be sure as I just bought this pixel 5a and immediately flashed the
newest image from Google then rooted.

**Comment by Mygod** [comment]
Disable hardware tethering acceleration, reboot and try again.

**Comment by sainz95** [comment]
Disabled and rebooted. Ran again with wire guard enabled and still not
working.

Latest log:
https://drive.google.com/file/d/14jwgMxeyIXRM1VWUNpQlvIpkmn_yye5s/view?usp=drivesdk

Thanks,

**Comment by timothyrowan** [comment]
Having the same issue on rooted Android 12 Pixel 4a 5g. Tried AdGuard DNS (FakeVPN) and WireGuard. Temp Hotspot is the only one that uses the device VPN. Regular Hotspot and USB Tether (preferred) skips the device VPN and IP check returns my cell carrier instead of VPN provider.

**Comment by sainz95** [comment]
I swapped to lineageos 18.1 and same issue unless I enable allow vpn
upstream for hotspot under the lineageos settings. That fixes the issue but
that toggle will not be present in native Android versions.

**Comment by Mygod** [comment]
Did you turn on the wlan1 switch in app?

**Reply by timothyrowan** [reply]
Using USB tethering to a GL iNet router. Tried AdGuard (Dummy VPN) last evening and switched on rndis0 after tethering was enabled. Seem to have done the trick. Checked my provider this AM and see that last night's data was mobile and not hotspot data. Thanks again for your help!
